src/ex1/script.js

```diff
@@ -0,0 +1,381 @@
+const TYPE_DEADLINE = 1
+const TYPE_NOTCOMPLETE = 2
+const TYPE_FINISHED = 3
+const TYPE_DEADLINE_FINISHED = 4
+const SORT_ALPHABATICALLY = 1;
+const SORT_DEADLINE = 2;
+const SORT_NOTCOMPLETE = 3;
+const SORT_FINISHED = 4;
+
+let TaskList = [];
+
+MainFunction();
+
+function MainFunction() {
+  AddTaskByEnter();
+  addTaskWatcher()
+}
+
+function addTaskWatcher() {
+  setInterval(() => {
+    let deadLineReachedTask = TaskList.filter(x => x.type == TYPE_DEADLINE).filter(x => x.datetime < new Date());
+    if (deadLineReachedTask && deadLineReachedTask.length > 0) {
+      let ids = deadLineReachedTask.map(x => x.id);
+      for (let index = 0; index < ids.length; index++) {
+        const liId = ids[index];
+        let task = getTask(liId);
+        updateTask(task.id, TYPE_DEADLINE_FINISHED, task.name, task.datetime, task.isAnimated);
+        bindTaskList(false);
+
+      }
+    }
+
+  }, 1000)
+}
+function addTask(type, name, date) {
+  let id = getId();
+  TaskList.push({ id: id, type: type, name: name, datetime: date, createDateTime: new Date(), isAnimated: true });
+  return id;
+}
+
+function updateTask(id, type, name, date, isAnimated) {
+  let indexFound = 0;
+  let createDateTime = new Date();
+  for (let index = 0; index < TaskList.length; index++) {
+    let element = TaskList[index];
+    if (element.id == id) {
+      createDateTime = element.createDateTime;
+      indexFound = index;
+      break;
+    }
+  }
+
+  TaskList[indexFound] = { id: id, type: type, name: name, datetime: date, createDateTime: createDateTime, isAnimated: isAnimated };
+}
+function getLastTaskId() {
+  TaskList.sort((a, b) => b.getTime() - a.getTime())[0].id;
+}
+function removeTask(id) {
+  TaskList = TaskList.filter(x => x.id != id);
+}
+
+
+function getTask(id) {
+  return TaskList.filter(x => x.id == id)[0]
+}
+
+function getId() {
+  if (TaskList.length == 0) {
+    return 1;
+  }
+  let latestId = TaskList.map(x => x.id).sort((p, c) => c - p);
+  return latestId[0] + 1;
+
+}
+
+function rotateImage() {
+  let randomNo = randomIntFromInterval(1, 75);
+  let imagePath = `images/${randomNo}.jpg`
+  document.getElementById('imageNBA').setAttribute('src', imagePath);
+  document.getElementsByClassName('NbaImage')[0].style.visibility = "visible"
+  const spinning = [
+    { transform: 'rotate(0deg) scale(0)' },
+    { transform: 'rotate(0) scale(1)' }
+
+  ];
+  const timing = {
+    duration: 2000,
+    iterations: 1,
+  }
+  document.getElementById('imageNBA').animate(spinning, timing);
+  setTimeout(() => {
+    const spinning = [
+      { transform: 'rotate(0) scale(1)' },
+      { transform: 'rotate(0deg) scale(0)' }
+    ];
+    const timing = {
+      duration: 2000,
+      iterations: 1,
+    }
+    document.getElementById('imageNBA').animate(spinning, timing);
+    setTimeout(() => { document.getElementsByClassName('NbaImage')[0].style.visibility = "hidden" }, 500)
+  }, 2000);
+}
+
+function CreateNewListItemElement() {
```

Comment:
```suggestion
function createNewListItemElement() {
```

(relevant to all variable and method names in javascript)